src/datasets/load.py

```diff
@@ -484,8 +484,12 @@ def get_module(self) -> MetricModule:
         raise NotImplementedError
 
 
-class CanonicalDatasetModuleFactory(_DatasetModuleFactory):
-    """Get the module of a canonical dataset. The dataset script is downloaded from GitHub."""
+class GithubDatasetModuleFactory(_DatasetModuleFactory):
+    """
+    Get the module of a dataset from GitHub (legacy).
+    The dataset script is downloaded from GitHub.
+    This class will eventually be removed and a HubDatasetModuleFactory will be used instead.
```

Comment:
I would add a `.. deprecated::` rst directive here: https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated 

With the new mdx docs, I don't know if this is still possible...

Reply:
I don't think we already have something dedicated to deprecation in the mdx docs. Anyway this class is not in the exported documentation so we can probably keep this this way